### PR TITLE
add recipe for jupysql

### PR DESCRIPTION
**Pull request by sterlinm**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by sterlinm** [comment]
I am willing to be listed as a maintainer.

**Comment by sterlinm** [comment]
@conda-forge/help-python, ready for review!

**Comment by sterlinm** [comment]
@conda-forge-admin, please ping conda-forge/help-python for review

File: recipes/jupysql/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/jupysql-{{ version }}.tar.gz
+  sha256: 8c328e18ff43f566905aaa8a26c7c4693d83be5ba5e69bd2bb6c1d6a3c3af865
+
+build:
+  script: {{ PYTHON }} -m pip install . -vv
```
Comment (dopplershift) [inline recipes/jupysql/meta.yaml:13]: This package is pure Python and can be noarch:
```suggestion
  script: {{ PYTHON }} -m pip install . -vv
  noarch: python
```

File: recipes/jupysql/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  run:
+    - python
+    - prettytable
+    - ipython >=1.0
+    - sqlalchemy <2
+    - sqlparse
+    - ipython_genutils >=0.1.0
+    - jinja2
+    - ploomber-core >=0.2.4
+    - importlib-metadata  # [py<38]
```
Comment (dopplershift) [inline recipes/jupysql/meta.yaml:29]: For noarch (better to just unconditionally depend on `importlib-metadata`):
```suggestion
    - python >=3.7
    - pip
  run:
    - python >=3.7
    - prettytable
    - ipython >=1.0
    - sqlalchemy <2
    - sqlparse
    - ipython_genutils >=0.1.0
    - jinja2
    - ploomber-core >=0.2.4
    - importlib-metadata
```

**Review by dopplershift — CHANGES_REQUESTED** [review]
Just some change to make this noarch.

**Comment by sterlinm** [comment]
Thanks so much! Applied both changes.

**Review by dopplershift — APPROVED** [review]

**Pull request merged**